### ISSUE TITLE
Access Request in the error handler

### ISSUE THREAD

**Issue by rkfox** [routerify/routerify#10]
Based on the examples, there does not appear to be a way to access the Request inside of the error handler. This means that error pages cannot be rendered using sessions.

**Comment by rousan** [comment]
@arkf0x, I'm planning to provide(for the next release) the request metadata to post middleware and the error handler as well.

The API will look like below:

```rust
async fn post_middleware_handler(res: Resonse<Body>, meta: RequestMeta) -> Result<Resonse<Body>, Error> {

// Do some work based on meta 

Ok(res)
}
```

The `RequestMeta` will have:
- Headers
- Uri
- Method

And also the error handler can access this as follows:

```rust
async fn my_err_handler(err: routerify::Error, meta: RequestMeta) -> Result<Resonse<Body>, Error>  {

// generate a response based on the meta.
}
```

**Comment by arkf0x** [comment]
Does this have an impact on performance? Just wondering if it make sense to be optional or not.

**Comment by rousan** [comment]
we have to make a benchmark on it, i don't think there will be any visible performance impact.

**Comment by rousan** [comment]
Solved in [v1.1.1](https://github.com/routerify/routerify/releases/tag/v1.1.1).

Now, error response can be generated based on the request info e.g. session managing.
An [example](https://github.com/routerify/routerify/blob/master/examples/error_handling_with_request_info.rs).
For more [info](https://docs.rs/routerify/1.1.1/routerify/index.html#error-handling-with-request-info).
